requirements.txt

```diff
@@ -1,6 +1,6 @@
-scikit-decide[all]==1.0.2
+scikit-decide==1.0.2
```

Comment:
We first need to understand the side effects of this because now many packages will need to be separately installed for many scripts of the challenge to work.

Reply:
I was about to include a comment about this: as far as I could test, everything works fine. With all the packages the docker image construction fails on my Mac

Reply:
Plus: with scikit-decide[all] we might get dangerously close to (or surpass) the 2GB limit for the docker image. With the regular scikit-decide version a submission image is already 1.23GB

Reply:
Ok, I made some tests again.

Everything works without the `[all]` specification, except for the `generate_solve_rllib_test.py` script, which has indeed very extensive dependencies.

To be honest I am uncomfortable with this: the entire toolkit is designed so that competitors are free to choose how they want to implement their solver. The RL testing script, however, requires by its nature more (and more opinionated) dependencies, which is understandable, but seems to run counter to the rest of the design.

As an additional problem, we are already too close for comfort to the size limit for the images to be sent; I am a bit afraid we might have passed that already with dependencies like PyTorch. This is a separate issue and deserves a dedicated discussion, but it's also one more reason to avoid forcing the installation of packages that some competitors may not want to use.

A simple solution might be to install by default the basic `scikit-decide` version, and explicitly tell the competitors that the RL example requires the `scikit-decide[all]` version.

What do you say?

Best

Reply:
As agreed in our call today, I replaced the `scikit-decide[all]` dependency with individual packages. I tested the RLIB script (the one that wasn't working before) both in a virtual environment and in docker on Apple silicon and it seems to work.